cores/esp8266/HardwareSerial.h

```diff
@@ -135,16 +135,45 @@ class HardwareSerial: public Stream
         // return -1 when data is unvailable (arduino api)
         return uart_peek_char(_uart);
     }
+
+    virtual bool peekBufferAPI () const override
+    {
+        return true;
+    }
+
+    // return a pointer to available data buffer (size = available())
+    // semantic forbids any kind of read() before calling peekConsume()
+    const char* peekBuffer () override
+    {
+        return uart_peek_buffer(_uart);
+    }
+
+    // return number of byte accessible by peekBuffer()
+    size_t availableForPeek () override
```

Comment:
This is pure style, so ignore as you want, but having all the peek APi start with `peek` seems like a good thing, so I'd really use `peekAvailable()` as the method name in the base class...

Reply:
> so I'd really use peekAvailable() as the method name in the base class...

OK